### DISCUSSION TITLE
Thoughts on polcal in crosscal and selfcal

### DISCUSSION THREAD

**Discussion by sphemakh** [caracal-pipeline/caracal#1550] [General]

**Comment by bennahugo** [comment]
Ok @PeterKamphuis @paoloserra as well as meerkathi admins you should have read access to https://github.com/ska-sa/fleeting-pol - our boresight polarization calibration pipeline for the tied array beamformer 

The relevant reports detailing MeerKAT leakage are the following:
 - https://docs.google.com/document/d/1Goa7jnDFR94hnnboM47d9p0-O0At5cNts5vj5nvO2fk/edit?usp=sharing
 - https://docs.google.com/document/d/19lSzq5Y1qYtNwgq3rWHfbxH0dHLGPiPwGxtXIDvOLpE/edit?usp=sharing
 - https://www.overleaf.com/read/yqyvmznvytcf 
 - https://docs.google.com/document/d/1PCcp4I_QxtVnEWr0Siv3Zt3wev3k9C3BzbE0CUyYfjc/edit?usp=sharing

The following needs to be edited in the pipeline:
  - crosscal: gain and bandpass calibration must apply P Jones on MODEL_DATA
  - crosscal: gain and bandpass calibration must use the same reference antenna as X, D, K and Kcross Jones 
  - polarization worker needs to be added which:
    - apply 90 degree FEED rotation using python-casacore to FEEDS table
    - first averages the data to improve SNR on X(chan) and D(chan) solutions
    - then compute crosshand delay (dominated by digitizer timestamp offsets between X and Y, on the Fengine data). This is the slope of X Jones on strongly linearly polarized source (when source crosshands are not 0 due to parallactic angle rotation)
    - then compute crosshand phase, X Jones. DC first, then per channel X(chan)
    - then compute delays, DC first, then per channel D(chan)
 - reports has to be migrated from fleetingpol which requires python-casacore as dependency
 - additional html report on crosshand equality needs to be compiled as described in [Polarization calibration testing procedures](https://docs.google.com/document/d/1Giv2JFYhnTm1wQX1xRp-LVeuVg_8atLXZgyXjfUvQLg/edit?usp=sharing) (company and stakeholder confidential) which I'm currently working on. I have the jupyter notebook ready - just needs to be incorporated in this pipeline.
 - then switch fleeting-pol to have meerkathi as dependency and act as wrapper for computing the jones corrections for the tied array beamformer

**Comment by gigjozsa** [comment]
@SpheMakh Can this be closed? It seems that polcal is now being implemented in #1099 , which is not this?

**Comment by Athanaseus** [comment]
Adding @francescaLoi to the discussion.

**Reply by francescaLoi** [reply]
I think this can be closed now @Athanaseus
